pkg/signals/signal.go

```diff
@@ -0,0 +1,30 @@
+package signals
+
+import (
+	"os"
+	"os/signal"
+)
+
+var (
+	onlyOneSignalHandler = make(chan struct{})
+	shutdownSignals      = []os.Signal{os.Interrupt}
+)
+
+// SetupSignalHandler registered for SIGTERM and SIGINT. A stop channel is returned
+// which is closed on one of these signals. If a second signal is caught, the program
```

Comment:
"If a second signal is caught, the program is terminated with exit code 1." --- this is a hardcoded rule that people don't expect in general. Do we really need it? (if the user really want to terminate it, he can send SIGKILL using command `kill -9`).

For graceful shutdown, I think typically it's done in the following way:

1. setup signal handler
1. start a goroutine that does the real work, block main function
1. when signal is caught, trigger graceful shutdown (e.g., closing a channel, cancel a context).
1. the main function will return normally when graceful shutdown is complete.

An example: https://gist.github.com/peterhellberg/38117e546c217960747aacf689af3dc2